### ISSUE TITLE
headerRight not accessible with Talkback screenreader (Android)

### ISSUE THREAD

**Issue by rosamerinoclaros** [react-navigation/react-navigation#12671]
### Current behavior

When using a `Stack.Screen` with both `headerRight` and `headerTitle` properties, the `headerRight` icon/button is inaccessible when TalkBack is enabled.

### Expected behavior

 The `headerRight` element should be reachable, clickable, and properly read by TalkBack.

### Reproduction

https://snack.expo.dev/@rosamerinoclaros/headerright-accessibility-bug

### Platform

- [x] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [x] @react-navigation/native-stack
- [ ] react-native-drawer-layout
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
    "react-native-gesture-handler": "2.25.0",
    "@react-navigation/bottom-tabs": "7.3.14",
    "@react-navigation/native": "7.1.10",
    "@react-navigation/native-stack": "7.3.14",
    "react-native-screens": "4.11.1",
    "react-native-reanimated": "3.17.4",
    "react-native": "0.79.2",
    "react": "19.0.0",

**Comment by rosamerinoclaros** [comment]
Upgrading those packages didn't help
